src/connector/src/sink/redis.rs

```diff
@@ -46,11 +49,73 @@ pub struct RedisCommon {
     #[serde(rename = "redis.url")]
     pub url: String,
 }
+pub enum RedisConn {
+    Cluster(ClusterConnection),
+    Single(MultiplexedConnection),
+}
+
+impl ConnectionLike for RedisConn {
+    fn req_packed_command<'a>(
+        &'a mut self,
+        cmd: &'a redis::Cmd,
+    ) -> redis::RedisFuture<'a, redis::Value> {
+        match self {
+            RedisConn::Cluster(conn) => conn.req_packed_command(cmd),
+            RedisConn::Single(conn) => conn.req_packed_command(cmd),
+        }
+    }
+
+    fn req_packed_commands<'a>(
+        &'a mut self,
+        cmd: &'a redis::Pipeline,
+        offset: usize,
+        count: usize,
+    ) -> redis::RedisFuture<'a, Vec<redis::Value>> {
+        match self {
+            RedisConn::Cluster(conn) => conn.req_packed_commands(cmd, offset, count),
+            RedisConn::Single(conn) => conn.req_packed_commands(cmd, offset, count),
+        }
+    }
+
+    fn get_db(&self) -> i64 {
+        match self {
+            RedisConn::Cluster(conn) => conn.get_db(),
+            RedisConn::Single(conn) => conn.get_db(),
+        }
+    }
+}
 
 impl RedisCommon {
-    pub(crate) fn build_client(&self) -> ConnectorResult<RedisClient> {
-        let client = RedisClient::open(self.url.clone())?;
-        Ok(client)
+    pub async fn build_conn(&self) -> ConnectorResult<RedisConn> {
+        let url: Value =
+            serde_json::from_str(&self.url).map_err(|e| SinkError::Config(anyhow!(e)))?;
```

Comment:
May add more details about the anyhow error, like `anyhow!("failed to parse url config: {:?}", e.as_report())`